wasm/validator/src/default.rs

```diff
@@ -0,0 +1,1000 @@
+use permission::{
+    validate_grant_account_permission, validate_register_permission_token,
+    validate_revoke_account_permission, Token as _,
+};
+use role::{validate_grant_account_role, validate_revoke_account_role, validate_unregister_role};
+use trigger::{
+    validate_execute_trigger, validate_mint_trigger_repetitions, validate_unregister_trigger,
+};
+use validator::validate_upgrade_validator;
+
+use super::*;
+use crate::{permission, prelude::*};
+
+macro_rules! custom_impls {
+    ( $($validator:ident($operation:ty)),+ $(,)? ) => { $(
+        #[allow(missing_docs)]
+        fn $validator(&mut self, authority: &AccountId, operation: $operation) -> Verdict {
+            $validator(self, authority, operation)
+        } )+
+    }
+}
+
+/// Apply `callback` macro for all token types from this crate.
+///
+/// Callback technique is used because of macro expansion order. With that technique we can
+/// apply callback to token types declared in other modules.
+///
+/// # WARNING !!!
+///
+/// If you add new module with tokens don't forget to add it here!
+macro_rules! map_all_crate_tokens {
+    ($callback:ident) => {
+        $crate::default::account::map_tokens!($callback);
+        $crate::default::asset::map_tokens!($callback);
+        $crate::default::asset_definition::map_tokens!($callback);
+        $crate::default::domain::map_tokens!($callback);
+        $crate::default::parameter::map_tokens!($callback);
+        $crate::default::peer::map_tokens!($callback);
+        $crate::default::role::map_tokens!($callback);
+        $crate::default::trigger::map_tokens!($callback);
+        $crate::default::validator::map_tokens!($callback);
+    };
+}
+
+macro_rules! tokens {
+    (
+        pattern = {
+            $(#[$meta:meta])*
+            $vis:vis struct _ {
+                $(
+                    $(#[$field_meta:meta])*
+                    $field_vis:vis $field:ident: $field_type:ty
+                ),* $(,)?
+            }
+        },
+        $module:ident :: tokens: [$($name:ident),+ $(,)?]
+    ) => {
+        declare_tokens!($(
+            crate::default::$module::tokens::$name
+        ),+);
+
+        pub mod tokens {
+            //! Permission tokens for concrete operations.
+
+            use super::*;
+
+            macro_rules! single_token {
+                ($name_internal:ident) => {
+                    $(#[$meta])*
+                    $vis struct $name_internal {
+                        $(
+                            $(#[$field_meta])*
+                            $field_vis $field: $field_type
+                        ),*
+                    }
+                };
+            }
+
+            $(single_token!($name);)+
+        }
+    };
+}
+
+pub(crate) use map_all_crate_tokens;
+pub(crate) use tokens;
+
+/// Trait which provides default validation
+///
+/// When implemented on a type it replaces some of [`Validate`]'s methods with sensible defaults
+pub trait DefaultValidator: Validate {
```

Comment:
Do we really need this `Validator` constraint here?

Reply:
we don't, but it seems to fit IMO